File: 1829-maximum-xor-for-each-query/1829-maximum-xor-for-each-query.py
class Solution:
    def getMaximumXor(self, nums: List[int], maximumBit: int) -> List[int]:
        # 전체 xor 하고, k xor 했을 때 최대 값. 
        # 이 k는 2^maximumbit 보다 작음
        # 가장 끝 숫자 제거
        n = len(nums)
        max_val = 2**maximumBit - 1
        tot = 0
        res = []
        for num in nums:
            tot ^= num
        for i in range(n-1, -1, -1):
            res.append(max_val ^ tot)
            tot ^= nums[i]
        return res

File: 1829-maximum-xor-for-each-query/1829-maximum-xor-for-each-query.rs
impl Solution {
    pub fn get_maximum_xor(nums: Vec<i32>, maximum_bit: i32) -> Vec<i32> {
        let mut tot = 0;
        let mut res = vec!();

        for num in &nums {
            tot ^= *num;
        }

        for num in nums.iter().rev() {
            res.push((!tot) & ((1 << maximum_bit) - 1));
            tot ^= *num;
        }
        res
    }
}
